parser/transaction.go

```diff
@@ -16,6 +16,10 @@ import (
 
 type rawTransaction struct {
 	fOverwintered      bool
+	isOverwinterV3     bool
+	isSaplingV4        bool
+	isZip225V5         bool
+	isGroth16Proof     bool // applicable to joinSplit only
```

Comment:
The `zcashd` transaction parser calculates these values internally to its parser in order to make parsing easier, but they are derived properties of `(fOverwintered, nVersionGroupId, version)`, not fields of `rawTransaction`. `zcashd` does not store those values on `CTransaction`.

So my concern here is, how is `rawTransaction` itself persisted? I'm reluctant to have these fields be part of any serialization format, because that means they can independently differ from `(fOverwintered, version)`.

If the current Go transaction parser structure is cached exactly as-is, then these fields should be replaced by methods that calculate them from `(fOverwintered, nVersionGroupId, version)`. (Or just pass the fields through into the internal partial parsers, like zcashd does.)